### PR TITLE
Allow customizable 'No accommodation' option in registration form

### PR DESCRIPTION
**Pull request by mvidalgarcia**
~I think a script similar to [this](https://gist.github.com/mvidalgarcia/b0fc94bd0a3b65a5dce0cb3b6a5cd92c) should be run to avoid errors in the existing registrations which have accommodation field.~
I think it's no necessary anymore.

image
image
image
image

### CODE REVIEW

File: indico/modules/events/registration/fields/choices.py [hunk 1]
```diff
@@ -358,13 +358,14 @@ def _stay_dates_valid(form, field):
...
                 return
             data = snakify_keys(field.data)
-            try:
-                arrival_date = data['arrival_date']
-                departure_date = data['departure_date']
-            except KeyError:
-                raise ValidationError(_("Arrival/departure date is missing"))
-            if _to_date(arrival_date) > _to_date(departure_date):
-                raise ValidationError(_("Arrival date can't be set after the departure date."))
+            if 'choice' in data and data['choice']:
```
Comment (ThiefMaster) [inline indico/modules/events/registration/fields/choices.py]: `if data.get('choice'):`

File: indico/modules/events/registration/fields/choices.py [hunk 2]
```diff
@@ -413,11 +414,12 @@ def process_form_data(self, registration, value, old_data=None, billable_items_l
             return {}
         data = {}
         if value:
-            data = {
-                'choice': value['choice'],
-                'arrival_date': value['arrivalDate'],
-                'departure_date': value['departureDate']
-            }
+            if 'choice' in value and value['choice']:
```
Comment (ThiefMaster) [inline indico/modules/events/registration/fields/choices.py]: same

File: indico/modules/events/registration/templates/display/_registration_summary_blocks.html [hunk 3]
```diff
@@ -23,22 +23,26 @@
...
-        </div>
-        <div>
-            {% trans accommodation=friendly_data['choice'] -%}
-                <strong>Accommodation:</strong> {{ accommodation }}
-            {%- endtrans %}
-        </div>
+    {% if field.input_type == 'accommodation' %}
+        {% if friendly_data %}
+            <div>
+                {% trans arrival_date=friendly_data['arrival_date'] | format_date -%}
```
Comment (ThiefMaster) [inline indico/modules/events/registration/templates/display/_registration_summary_blocks.html]: I know it was like this before, but it'd be a good opportunity to change this to `friendly_data.arrival_date` (same for the other ones below)

File: indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/base.tpl.html [hunk 4]
```diff
@@ -38,7 +38,7 @@
                         ng-class="{hasError: validationStarted && optionsForm.title.$invalid}"
                         nd-init-focus
                         class="caption-input"/>
-                    <span ng-if="settings.isRequired">
+                    <span ng-if="field.inputType != 'accommodation'">
```
Comment (OmeGak) [inline indico/htdocs/js/indico/modules/registration/form/tpls/fields/dialogs/base.tpl.html]: I don't think we want this. `accommodation` is not the only field we want to disallow making mandatory. With this change, for instance, `label` fields will be allowed to make mandatory, which doesn't make any sense.

Instead, what we need to do is adding `settings.isRequired = false;` in the `link` method of `ndAccommodationField`.
https://github.com/indico/indico/blob/v1.9.9/indico/htdocs/js/indico/modules/registration/form/field.js#L953

In the same way we do here:
https://github.com/indico/indico/blob/v1.9.9/indico/htdocs/js/indico/modules/registration/form/field.js#L444

**Comment by OmeGak** [comment]
In general it is better to split linter-only fixes into a standalone PR. Otherwise, it makes it difficult to focus on functional changes while reviewing. This PR is small and didn't cause much trouble, but keep it mind.

**Comment by mvidalgarcia** [comment]
@OmeGak: Not even if they are in a separated commit as here? Should I create a new PR each time I have to change just a couple of things related to linter?

**Comment by ThiefMaster** [comment]
You could even ignore them if they are in semi-legacy code. Or ping someone to cherrypick the commit - no need for a PR!

File: indico/modules/events/registration/templates/display/_registration_summary_blocks.html [hunk 5]
```diff
@@ -24,16 +24,18 @@
...
-            {% trans arrival_date=friendly_data['arrival_date'] | format_date -%}
-                <strong>Arrival:</strong> {{ arrival_date }}
-            {%- endtrans %}
-        </div>
-        <div>
-            {% trans departure_date=friendly_data['departure_date'] | format_date -%}
-                <strong>Departure:</strong> {{ departure_date }}
-            {%- endtrans %}
-        </div>
+         {% if not friendly_data['is_no_accommodation'] %}
```
Comment (pferreir) [inline indico/modules/events/registration/templates/display/_registration_summary_blocks.html]: `friendly_data.is_no_accommodation`

File: indico/htdocs/js/indico/modules/registration/form/field.js [hunk 6]
```diff
@@ -15,6 +15,8 @@
  * along with Indico; if not, see <http://www.gnu.org/licenses/>.
  */
 
+/*globals angular, ndRegForm */
```
Comment (ThiefMaster) [inline indico/htdocs/js/indico/modules/registration/form/field.js]: /* global angular, ndRegForm */

**Pull request merged**